components/role-mgt/org.wso2.carbon.identity.role.v2.mgt.core/src/main/java/org/wso2/carbon/identity/role/v2/mgt/core/RoleManagementService.java

```diff
@@ -423,23 +422,37 @@ List<String> getRoleIdListOfIdpGroups(List<String> groupIds, String tenantDomain
     /**
      * Delete all roles associated app by id.
      *
-     * @param applicationId        Idp Group IDs.
-     * @param tenantDomain    Tenant domain.
+     * @param applicationId Idp Group IDs.
+     * @param tenantDomain  Tenant domain.
      * @throws IdentityRoleManagementException IdentityRoleManagementException.
      */
     void deleteRolesByApplication(String applicationId, String tenantDomain) throws IdentityRoleManagementException;
 
     /**
      * Get main role to shared role mappings by subOrg.
      *
-     * @param roleIds        Main role IDs.
+     * @param roleIds            Main role IDs.
      * @param subOrgTenantDomain Sub Organization tenant domain.
      * @return The map of main role id to shared roles.
      * @throws IdentityRoleManagementException IdentityRoleManagementException.
      */
     Map<String, String> getMainRoleToSharedRoleMappingsBySubOrg(List<String> roleIds, String subOrgTenantDomain)
             throws IdentityRoleManagementException;
 
+    /**
+     * Get shared role to main role mappings by subOrg.
+     *
+     * @param roleIds            Shared role IDs.
+     * @param subOrgTenantDomain Sub Organization tenant domain.
+     * @return The map of shared role id to main roles.
+     * @throws IdentityRoleManagementException IdentityRoleManagementException.
+     */
+    default Map<String, String> getSharedRoleToMainRoleMappingsBySubOrg(List<String> roleIds, String subOrgTenantDomain)
+            throws IdentityRoleManagementException {
+
+        throw new NotImplementedException("getRoles method is not implemented");
```

Comment:
```suggestion
        throw new NotImplementedException("getSharedRoleToMainRoleMappingsBySubOrg method is not implemented");
```

Reply:
Addressed and resolved in commit: [cccc26e](https://github.com/wso2/carbon-identity-framework/pull/6162/commits/cccc26ee039e448196fc1299e2827453c1bc76b5)